### PR TITLE
Makes the gem compatible with Rails 5

### LINKED ISSUE #13: Add support for Rails 5.0
Simply adding Rails 5.0 to dependency leads to broken tests.

### PR DESCRIPTION
**Pull request by walski**
Fixes #13 

I must admit I have not the slightest clue about Arel and AR internals but the specs pass again and it seems to work :)

Happy to apply more changes as you find necessary @take-five

### CODE REVIEW

**Comment by take-five** [comment]
Hi @walski 
Thanks for the PR, I don't really have time to work on this project right now.
Unfortunately all the existing specs for Rails 3.x and 4.x have failed, but I think it'll be better to have version for Rails 5 in a separate branch and get rid of all the nasty hacks that keep the gem compatible with older Rails.

**Comment by take-five** [comment]
Could you change your PR destination to `rails-5` branch instead of `master`?

**Comment by walski** [comment]
There we go :)

File: spec/active_record/hierarchical_query_spec.rb [hunk 1]
```diff
@@ -218,19 +218,6 @@ def assert_ordered_by_name_asc(&block)
         ).to match_array [child_4, child_5]
       end
     end
-
-    describe 'binding values' do
```
Comment (take-five) [inline spec/active_record/hierarchical_query_spec.rb]: I wonder, why this spec is removed?

Comment (walski) [inline spec/active_record/hierarchical_query_spec.rb]: It seems like the ``bind`` method is just gone from AR 5:

* https://github.com/rails/rails/blob/master/activerecord/lib/active_record/relation/query_methods.rb
* https://apidock.com/rails/ActiveRecord/QueryMethods/bind

So my conclusion was: Test is not necessary anymore :) I've also removed the ``bind`` method from the ``CHILD_SCOPE_METHODS`` in ``query.rb`` due to that. Other ideas?

**Comment by take-five** [comment]
Thanks a lot!

**Comment by take-five** [comment]
Check out version 1.0.0 from rubygems.org :)

**Pull request merged**